### ISSUE TITLE
Significantly slower speeds when using NGINX/Clourflared tunnels

### ISSUE THREAD

**Issue by brendongl** [a1ex4/ownfoil#161]
This may not be related to ownfoil.
But Ive tested a shop with these different configurations

sub-domain proxied via Cloudflared Zero Trust Tunnel
sub-domain via Nginx Proxy Manager via Cloudflare
ddns via Router with port-forwarded
current WAN IP address with port-forwarded
Local IP with port

The DL speed i get with the cloudflared/nginx manager options are usually between 1 to 2 MB/s
While with WAN IP+port forwarding and Local IP were closer together at 8-12MB/s

Can anyone help with where im going wrong with settings etc?
I think I would rather use Nginx as it seems to be a better option that cloudflared.
I'm using Unraid with the latest commit.

I see
"Configure your shop URL to enable host verification.
Host verification from Tinfoil requests will be enforced if the shop is accessed securely (with https), make sure your reverse proxy does NOT allow insecure connections (or upgrades them to use SSL) and correctly sets the X-Forwarded-Proto header. "

I add the shop URL into that field but it doesnt seem to be doing anything - im not using HTTPS protocol.

**Comment by a1ex4** [comment]
Yes that's how the internet works, locally in your LAN you are only limited by the speed of your internal network, so between your router and your devices. With remote host you are limited first by your upload speed to Cloudflare servers, and by your download speed to those same servers.
